src/commands/coverage/upload.ts

```diff
@@ -0,0 +1,356 @@
+import fs from 'fs'
+import os from 'os'
+import path from 'path'
+
+import chalk from 'chalk'
+import {Command, Option} from 'clipanion'
+import glob from 'glob'
+import * as t from 'typanion'
+
+import {getCISpanTags} from '../../helpers/ci'
+import {getGitMetadata} from '../../helpers/git/format-git-span-data'
+import {RequestBuilder, SpanTags} from '../../helpers/interfaces'
+import {Logger, LogLevel} from '../../helpers/logger'
+import {retryRequest} from '../../helpers/retry'
+import {GIT_REPOSITORY_URL, GIT_SHA, parseMetrics, parseTags} from '../../helpers/tags'
+import {getUserGitSpanTags} from '../../helpers/user-provided-git'
+import {buildPath, getRequestBuilder, timedExecAsync} from '../../helpers/utils'
+
+import {isGitRepo} from '../git-metadata'
+import {newSimpleGit} from '../git-metadata/git'
+import {uploadToGitDB} from '../git-metadata/gitdb'
+
+import {apiConstructor, apiUrl, intakeUrl} from './api'
+import id from './id'
+import {APIHelper, Payload} from './interfaces'
+import {
+  renderCommandInfo,
+  renderDryRunUpload,
+  renderFailedGitDBSync,
+  renderFailedUpload,
+  renderInvalidFile,
+  renderRetriedUpload,
+  renderSuccessfulGitDBSync,
+  renderSuccessfulUpload,
+  renderSuccessfulUploadCommand,
+  renderUpload,
+} from './renderer'
+import {detectFormat, isFile, validateCoverageReport} from './utils'
+
+const TRACE_ID_HTTP_HEADER = 'x-datadog-trace-id'
+const PARENT_ID_HTTP_HEADER = 'x-datadog-parent-id'
+const errorCodesStopUpload = [400, 403]
+
+const MAX_REPORTS_PER_REQUEST = 10
+
+export class UploadCodeCoverageReportCommand extends Command {
+  public static paths = [['coverage', 'upload']]
+
+  public static usage = Command.Usage({
+    category: 'CI Visibility',
+    description: 'Upload code coverage reports files to Datadog.',
+    details: `
+      This command will upload code coverage report files to Datadog.\n
+      See README for details.
+    `,
+    examples: [
+      ['Upload all code coverage report files in current directory', 'datadog-ci coverage upload --path .'],
+      ['Send code coverage upload completed signal', 'datadog-ci coverage upload --flush'],
+      [
+        'Upload all code coverage report files in current directory and send code coverage upload completed signal',
+        'datadog-ci coverage upload --path . --flush',
+      ],
+      [
+        'Upload all code coverage report files in src/unit-test-coverage and src/acceptance-test-coverage',
+        'datadog-ci coverage upload --path src/unit-test-coverage --path src/acceptance-test-coverage',
+      ],
+      [
+        'Upload all code coverage report files in current directory and add extra tags globally',
+        'datadog-ci coverage upload --tags key1:value1 --tags key2:value2 --path .',
+      ],
+      [
+        'Upload all code coverage report files in current directory and add extra measures globally',
+        'datadog-ci coverage upload --measures key1:123 --measures key2:321 .',
```

Comment:
Did you forget `--path` here? I agree with @juan-fernandez that it would probably be better for `--path` to be positional (see [Clipanion docs](https://mael.dev/clipanion/docs/options#rests)):

```sh
datadog-ci coverage upload src/unit-test-coverage src/acceptance-test-coverage
```

Reply:
Good catch. Changed the `--path` to be a positional argument